Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -368,7 +370,9 @@ void readable_stream_default_reader_read(ReadableStreamDefaultReader& reader, Re
         VERIFY(stream->is_readable());
 
         // 2. Perform ! stream.[[controller]].[[PullSteps]](readRequest).
-        MUST(stream->controller()->pull_steps(read_request));
+        MUST(stream->controller()->visit([&](auto const& controller) {
```

Comment:
The use of `TRY` and `MUST` for these seems pretty arbitrary, please explain with a comment

Reply:
True, I think it's the `!` that confused me. Just to be sure, `TRY` here is correct because, even though `pull_steps` can't throw a JS exception, it can throw OOM right? Seems like `TRY/MUST_OR_THROW_OOM` would make more sense here, but neither of those work as I believe they both want a `Completion`

Reply:
That sounds right, it's up to the caller to check whether downstream a `!` call could still throw OOM. I'll refactor a bunch of these to use `ErrorOr` after this PR is merged, which should be easier to reason about (see https://github.com/SerenityOS/serenity/blob/master/Documentation/Browser/Patterns.md, 'Error Handling')